psutil/_psutil_sunos.c

```diff
@@ -139,7 +190,21 @@ psutil_proc_name_and_args(PyObject *self, PyObject *args) {
     py_name = PyUnicode_DecodeFSDefault(info.pr_fname);
     if (!py_name)
         goto error;
-    py_args = PyUnicode_DecodeFSDefault(info.pr_psargs);
+
```

Comment:
Please add a comment explaining what this section is for, e.g. that you first attempt reading process memory then parse /proc which may truncate args.